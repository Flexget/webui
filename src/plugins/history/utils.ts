import { groupBy } from 'utils/array';
import { HistoryItem, GroupByFields } from './types';

export const getGroupedHistory = (items: HistoryItem[], grouping: GroupByFields) =>
  groupBy(
    items,
    grouping,
    grouping === GroupByFields.Time ? (i: string) => i.split('T')[0] : undefined,
  );
