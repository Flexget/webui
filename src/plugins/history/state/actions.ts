import statusActions from 'core/status/state/actions';
import { StatusError } from 'utils/fetch';
import { action, ActionsUnion } from 'utils/actions';
import { RequestsOfType as ROT } from 'core/status/state/types';
import { GetHistoryOptions, History } from './types';

export const enum Constants {
  GET_HISTORY = '@flexget/history/GET_HISTORY',
}

const actions = {
  getHistory: {
    request: (data: GetHistoryOptions) => statusActions.load(Constants.GET_HISTORY, data),
    success: (data: History[], headers: Headers, refresh: boolean) =>
      action(Constants.GET_HISTORY, { data, headers, refresh }),
    failure: (err: StatusError) => statusActions.error(Constants.GET_HISTORY, err),
  },
};

export type ActionTypes = ActionsUnion<typeof actions>;
export type RequestsOfType<T extends Constants> = ROT<ActionTypes, T>;

export default actions;
