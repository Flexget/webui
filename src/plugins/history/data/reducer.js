import { GET_HISTORY } from 'plugins/history/data/actions';

const initState = {
  totalCount: 0,
  items: [],
};


export default function reducer(state = initState, { type, payload }) {
  switch (type) {
    case GET_HISTORY:
      return {
        totalCount: payload.headers.get('total-count'),
        items: payload.refresh ? payload.data : [...state.items, ...payload.data],
      };
    default:
      return state;
  }
}
