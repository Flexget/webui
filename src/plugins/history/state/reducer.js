import { GET_HISTORY } from './actions';

const initState = {
  items: [],
};


export default function reducer(state = initState, { type, payload }) {
  switch (type) {
    case GET_HISTORY:
      return {
        totalCount: payload.headers.get('total-count'),
        items: payload.refresh ? payload.data : [...state.items, ...payload.data],
      };
    default:
      return state;
  }
}
