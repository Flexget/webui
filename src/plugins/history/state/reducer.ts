import { Constants, ActionTypes } from './actions';
import { History } from '../types';

export interface State {
  items: History[];
  totalCount?: number;
}

const initState: State = {
  items: [],
};

export default function reducer(state = initState, action: ActionTypes): State {
  switch (action.type) {
    case Constants.GET_HISTORY: {
      const { payload } = action;
      return {
        totalCount: parseInt(payload.headers.get('total-count') ?? '0', 10),
        items: payload.refresh ? payload.data : [...state.items, ...payload.data],
      };
    }
    default:
      return state;
  }
}
