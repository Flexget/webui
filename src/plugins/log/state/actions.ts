import { action, ActionsUnion } from 'utils/actions';
import statusActions from 'core/status/state/actions';
import { RequestsOfType as ROT } from 'core/status/state/types';
import { StatusError } from 'utils/fetch';
import { LogMessage } from '../types';

export const enum Constants {
  LOG_CONNECT = '@flexget/log/LOG_CONNECT',
  LOG_MESSAGE = '@flexget/log/LOG_MESSAGE',
  LOG_DISCONNECT = '@flexget/log/LOG_DISCONNECT',
  LOG_CLEAR = '@flexget/log/LOG_CLEAR',
}

interface ConnectParams {
  query: string;
  lines: string;
}

const actions = {
  connect: {
    request: (params: ConnectParams) => statusActions.load(Constants.LOG_CONNECT, params),
    success: () => action(Constants.LOG_CONNECT),
    failure: (err: StatusError) => statusActions.error(Constants.LOG_CONNECT, err),
  },
  message: (message: LogMessage) => action(Constants.LOG_MESSAGE, message),
  disconnect: () => action(Constants.LOG_DISCONNECT),
  clear: () => action(Constants.LOG_CLEAR),
};

export type ActionTypes = ActionsUnion<typeof actions>;
export type RequestsOfType<T extends Constants> = ROT<ActionTypes, T>;

export default actions;
