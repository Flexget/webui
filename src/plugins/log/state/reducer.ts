import { Constants, ActionTypes } from './actions';
import { LogMessage } from './types';

interface State {
  messages: LogMessage[];
  connected: Boolean;
}

const initState: State = {
  messages: [],
  connected: false,
};

export default function reducer(state = initState, action: ActionTypes): State {
  switch (action.type) {
    case Constants.LOG_CONNECT:
      return {
        ...state,
        messages: [],
        connected: true,
      };
    case Constants.LOG_MESSAGE:
      return {
        ...state,
        messages: [action.payload, ...state.messages],
      };
    case Constants.LOG_DISCONNECT:
      return {
        ...state,
        connected: false,
      };
    case Constants.LOG_CLEAR:
      return {
        ...state,
        messages: [],
      };
    default:
      return state;
  }
}
