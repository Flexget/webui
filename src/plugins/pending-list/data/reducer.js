import * as actions from './actions';

const initState = {
  lists: [],
  entries: {},
};

export default function reducer(state = initState, { payload, type }) {
  switch (type) {
    case actions.GET_LISTS:
      return {
        ...state,
        lists: payload,
      };
    case actions.ADD_LIST:
      return {
        ...state,
        lists: [...state.lists, payload],
      };
    case actions.REMOVE_LIST:
      return {
        ...state,
        lists: state.lists.filter(item => item.id === payload.id),
      };
    case actions.GET_ENTRIES:
      return {
        ...state,
        entries: {
          ...state.entries,
          [payload.listId]: payload.entries,
        },
      };
    case actions.ADD_ENTRY:
      return {
        ...state,
        entries: {
          ...state.entries,
          [payload.listId]: [
            ...state.entries[payload.listId],
            payload.entry,
          ],
        },
      };
    case actions.REMOVE_ENTRY:
      return {
        ...state,
        entries: {
          ...state.entries,
          [payload.listId]: state.entries[payload.listId].filter(item => item.id === payload.id),
        },
      };
    case actions.REJECT_ENTRY:
    case actions.APPROVE_ENTRY:
      return {
        ...state,
        entries: {
          ...state.entries,
          [payload.listId]: state.entries[payload.listId].map((item) => {
            if (item.id === payload.entry.id) {
              return payload.entry;
            }
            return item;
          }),
        },
      };
    default:
      return state;
  }
}
