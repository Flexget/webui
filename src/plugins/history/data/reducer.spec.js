import reducer from 'plugins/history/data/reducer';
import { GET_HISTORY } from 'plugins/history/data/actions';
import { Headers } from 'utils/tests';


describe('plugins/history/data/reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should store history on GET_HISTORY', () => {
    expect(reducer(undefined, {
      type: GET_HISTORY,
      payload: {
        data: [{ an: 'object' }],
        headers: new Headers({ 'total-count': 1 }),
      },
    })).toMatchSnapshot();
  });
});
