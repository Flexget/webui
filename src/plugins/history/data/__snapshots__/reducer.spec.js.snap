// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugins/history/data/reducer should return the initial state 1`] = `
Object {
  "items": Array [],
  "totalCount": 0,
}
`;

exports[`plugins/history/data/reducer should store history on GET_HISTORY 1`] = `
Object {
  "items": Array [
    Object {
      "an": "object",
    },
  ],
  "totalCount": 1,
}
`;
